src/calliope/util/tools.py

```diff
@@ -47,3 +51,52 @@ def listify(var: Any) -> list:
     else:
         var = [var]
     return var
+
+
+def climb_template_tree(
+    dim_item_dict: "AttrDict",
+    templates: "AttrDict",
+    item_name: str,
+    inheritance: list | None = None,
+) -> tuple["AttrDict", list | None]:
+    """Follow the `template` references from model definition elements to `templates`.
+
+    Model definition elements can inherit template entries (those in `templates`).
+    Template entries can also inherit each other, to create an inheritance chain.
+
+    This function will be called recursively until a definition dictionary without `template` is reached.
```

Comment:
The wording here in the docstring suggests that the function calls itself recursively, but that doesn't seem to actually be the case.

Reply:
It is the case, there is a call to `climb_template_tree` inside `climb_template_tree`.